cogeo_mosaic/backend/base.py

```diff
@@ -0,0 +1,31 @@
+"""cogeo_mosaic.backend.base: base Backend class."""
+
+import abc
+import contextlib
+from typing import Dict
+
+
+class BaseBackend(metaclass=contextlib.AbstractContextManager):
+    @abc.abstractmethod
+    def __init__(self, *args, **kwargs):
+        """Load resource"""
+
+    def metadata(self) -> Dict:
+        """Retrieve MosaicJSON metadata."""
+        return self.mosaic_def
```

Comment:
as mentioned before, this makes thing not really clear if metadata is returning metadata about the mosaic or the full mosaicjson document

Reply:
I'm not sure the best way to solve that? The difference is whether `tiles` has the full array or not?

Reply:
I think maybe it would be best to never return `tiles`, or at least default to not...
So
```py
def metadata(self, tiles=False) -> Dict:
	"""Retrieve Mosaic metadata

	Args:
	- tiles: Include `tiles` key in response

	Returns: MosaicJSON as dict, not including the `tiles` key unless specified
	"""
	if tiles:
		if 'tiles' not in self.mosaic_def.keys():
			raise ValueError...
		return self.mosaic_def

	return {k: v for k, v in self.mosaic_def.items() if k != 'tiles'}
```

This way

1. It can be consistent across all backends, even for potential new backends  in the future where it's also hard to get a list of all keys
2. Minimizes default return payload size, since as we've seen including `tiles` can make the full mosaicjson huge.

Reply:
> I think maybe it would be best to never return tiles

That's what I meant

Reply:
Ok, I misunderstood, but I think now it's fixed as you wanted?

Reply:
yeah I still have strong opinion about `metadata` should return metadata of the mosaic definition not the tiles (even with an option) but we can revisit this later

Reply:
I suppose you're right... What are the use cases for calling `metadata`? 
Essentially only the `tile()` and `point()` methods would need to access `tiles`, right?

Reply:
Ok I updated to never return `tiles`